pkg/testframework/console-operator.go

```diff
@@ -210,3 +210,51 @@ func SetLogLevel(t *testing.T, client *Clientset, logLevel operatorsv1.LogLevel)
 func GenerationChanged(oldGeneration, newGeneration int64) bool {
 	return oldGeneration == newGeneration
 }
+
+// the operator is settled if conditions match:
+// - available: true
+// - progressing: false
+// - degraded: false
+func operatorIsSettled(operatorConfig *operatorsv1.Console) bool {
+	settled := true
+	for _, condition := range operatorConfig.Status.Conditions {
+		if condition.Type == operatorsv1.OperatorStatusTypeAvailable {
+			if condition.Status == operatorsv1.ConditionFalse {
+				settled = false
+			}
+		}
+		if condition.Type == operatorsv1.OperatorStatusTypeProgressing {
+			if condition.Status == operatorsv1.ConditionTrue {
+				settled = false
+			}
+		}
+		if condition.Type == operatorsv1.OperatorStatusTypeDegraded {
+			if condition.Status == operatorsv1.ConditionTrue {
+				settled = false
+			}
+		}
+	}
+	return settled
+}
+
+// A helper to ensure our operator config reaches a settled state before we
+// begin the next test.
+func WaitForSettledState(t *testing.T, client *Clientset) (settled bool, err error) {
+	interval := 1 * time.Second
+	max := 20 * time.Second
+	count := 0
```

Comment:
not sure whats the point of the `count` variable. `wait.Poll` should not need it

Reply:
I'll remove.  I was logging it to see how many iterations a typical test took to pass.  It was interesting (some definitely take longer to settle than others), but not needed in the end.